File: practica2_839756_842236/c++/instruccionOver.cc
// Autores: Andrei Vlasceanu [839756] & Andres Yubero [842236]

#include "instruccionOver.h"

void InstruccionOver::ejecutar(stack<int> *s)
{
    int x, y;
    x = s->top();
    s->pop();
    y = s->top();
    s->pop();
    s->push(y);
    s->push(x);
    s->push(y);
}

void InstruccionOver::listar()
{
    cout << "over" << endl;
}

File: practica2_839756_842236/java/InstruccionOver.java
import java.util.Stack;

public class InstruccionOver extends Instruccion {
    @Override
    public void ejecutar(Stack<Integer> s) {
        int x = s.pop();
        int y = s.pop();
        s.push(y);
        s.push(x);
        s.push(y);
    }

    @Override
    public void listar() {
        System.out.println("over");
    }
}
